app/models/metric/helper.rb

```diff
@@ -120,9 +120,7 @@ def self.sanitize_start_end_time(interval, interval_name, start_time, end_time)
   end
 
   def self.remove_duplicate_timestamps(recs)
-    if recs.respond_to?(:klass) # active record relation
-      return recs unless recs.klass.kind_of?(Metric) || recs.klass.kind_of?(MetricRollup)
-    elsif recs.empty? || !recs.all? { |r| r.kind_of?(Metric) || r.kind_of?(MetricRollup) }
+    if recs.empty? || !recs.all? { |r| r.kind_of?(Metric) || r.kind_of?(MetricRollup) }
```

Comment:
I think we can still use previous code and only change
this condition:
https://github.com/ManageIQ/manageiq/blob/master/app/models/metric/helper.rb#L124
to
```ruby
 return recs unless recs.klass <= Metric || recs.klass <=MetricRollup
```

because `kind_of?` is for instances and not for classes.

I think that with this change we can fix the bug and keep @kbrock's intention in https://github.com/ManageIQ/manageiq/pull/14224.

it should work also with `VmPerformance` because it is inherited from `MetricRollup`.

It would be also nice to add specs when the methods has AR scope as input (`MetricRollup.where(..)`)

thanks!

Reply:
Yep! Done